mesonbuild/cmake/generator.py

```diff
@@ -67,8 +66,7 @@ def target_property(arg: str) -> str:
 
     def target_file(arg: str) -> str:
         if arg not in trace.targets:
-            mlog.warning(f"Unable to evaluate the cmake variable '$<TARGET_FILE:{arg}>'.")
-            return ''
+            return arg
```

Comment:
Technically, just returning arg here is wrong. (Since it will return `cm_exe_local` instead of the actual location of the newly built exe). This works in this case, since `add_custom_command` also accepts a target name as the first argument.

So, while returning `arg` is technically wrong, I would argue that it is "less wrong" than returning an empty string, since it works in *some* cases (but will still potentially fail for others).

However, the warning should not be removed, since we are still guessing and future issues will be easier to identify (mayebe reword it to `Unable to evaluate the...` --> `Unable to correctly evaluate the...`).

Reply:
Done